python/cudf/cudf/core/series.py

```diff
@@ -6448,6 +6448,30 @@ def is_month_start(self):
         """
         return (self.day == 1).fillna(False)
 
+    @property
+    def is_year_start(self):
+        """
+        Boolean indicator if the date is the first day of the month.
+
+        Returns
+        -------
+        Series
+        Booleans indicating if dates are the first day of the month.
+        """
+        return (self.day_of_year == 1).fillna(False)
+
+    @property
+    def is_year_end(self):
+        """
+        Boolean indicator if the date is the first day of the month.
+
+        Returns
+        -------
+        Series
+        Booleans indicating if dates are the first day of the month.
+        """
+        return (self.day_of_year == 365).fillna(False)
```

Comment:
does this work for leap years?

Reply:
You're right, this does not work for leap years. Adding to the code to factor that in.